src/libcore/cmp.rs

```diff
@@ -107,67 +108,29 @@ use self::Ordering::*;
 ///     format: BookFormat,
 /// }
 ///
+/// // Implement <Book> == <BookFormat> comparisons
 /// impl PartialEq<BookFormat> for Book {
 ///     fn eq(&self, other: &BookFormat) -> bool {
-///         match (&self.format, other) {
-///            (BookFormat::Paperback, BookFormat::Paperback) => true,
-///            (BookFormat::Hardback,  BookFormat::Hardback)  => true,
-///            (BookFormat::Ebook,     BookFormat::Ebook)     => true,
-///            (_, _) => false,
-///         }
+///         self.format == *other
 ///     }
 /// }
 ///
-/// let b1 = Book { isbn: 3, format: BookFormat::Paperback };
-///
-/// assert!(b1 == BookFormat::Paperback);
-/// assert!(b1 != BookFormat::Ebook);
-/// ```
-///
-/// By changing `impl PartialEq for Book` to `impl PartialEq<BookFormat> for Book`,
-/// we've changed what type we can use on the right side of the `==` operator.
-/// This lets us use it in the `assert!` statements at the bottom.
-///
-/// You can also combine these implementations to let the `==` operator work with
-/// two different types:
-///
-/// ```
-/// enum BookFormat {
-///     Paperback,
-///     Hardback,
-///     Ebook,
-/// }
-///
-/// struct Book {
-///     isbn: i32,
-///     format: BookFormat,
-/// }
-///
-/// impl PartialEq<BookFormat> for Book {
```

Comment:
I think the purpose of the sample you removed was to show off that you can `impl PartialEq for Book` multiple times by changing the type parameter, which allows you to do `<Book> == <Book>` *and* `<Book> == <BookFormat>`. If i remember right, we had a question about that a while back, which is why this was added.

Reply:
fair enough. readded in https://github.com/rust-lang/rust/pull/57357/commits/1445a065ded1bc7e73e7a358f9bc971e908e9181